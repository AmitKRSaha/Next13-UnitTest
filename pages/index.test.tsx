import { render, waitFor } from "@testing-library/react";
import Home from "./index";

import * as service from "./api.service";
jest.mock("./api.service");

// ============ Test file passing successfully using npm test but not in Jest extension ===========
describe("index test", () => {
  beforeEach(() => {
    jest.spyOn(service, "getTodos").mockResolvedValue([
      {
        userId: 1,
        id: 1,
        title: "delectus aut autem",
        completed: false,
      },
      {
        userId: 1,
        id: 2,
        title: "quis ut nam facilis et officia qui",
        completed: false,
      },
    ]);
  });

  test("render component",async () => {
    const { container } = render(<Home />);
    await waitFor(() => expect(service.getTodos).toHaveBeenCalledTimes(1))
    expect(container).toMatchSnapshot();
  });
});
