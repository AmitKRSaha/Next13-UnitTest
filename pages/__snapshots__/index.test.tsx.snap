// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index test render component 1`] = `
<div>
  <div>
    Next js Unit Test
    <div>
      delectus aut autem
    </div>
    <div>
      quis ut nam facilis et officia qui
    </div>
  </div>
</div>
`;
